Civi/RemoteTools/JsonSchema/FormSpec/Factory/FileFieldFactory.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/*
+ * Copyright (C) 2024 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\RemoteTools\JsonSchema\FormSpec\Factory;
+
+use Civi\RemoteTools\Form\FormSpec\AbstractFormField;
+use Civi\RemoteTools\Form\FormSpec\Field\FileField;
+use Civi\RemoteTools\JsonSchema\JsonSchema;
+use Civi\RemoteTools\JsonSchema\JsonSchemaNull;
+use Civi\RemoteTools\JsonSchema\JsonSchemaObject;
+use Civi\RemoteTools\JsonSchema\JsonSchemaString;
+use Civi\RemoteTools\Util\ArrayUtil;
+use Civi\RemoteTools\Util\FormatUtil;
+use CRM_Remotetools_ExtensionUtil as E;
+use Webmozart\Assert\Assert;
+
+/**
+ * @phpstan-import-type TValue from JsonSchema
+ */
+final class FileFieldFactory extends AbstractFieldJsonSchemaFactory {
+
+  public static function getPriority(): int {
+    return IntegerFieldFactory::getPriority() + 1;
+  }
+
+  public function createSchema(AbstractFormField $field): JsonSchema {
+    Assert::isInstanceOf($field, FileField::class);
+    /** @var \Civi\RemoteTools\Form\FormSpec\Field\FileField $field */
+
+    $properties = [
+      'filename' => new JsonSchemaString(['minLength' => 1, 'maxLength' => 255]),
+    ];
+    $keywords = [
+      'required' => ['filename'],
+    ];
+
+    $contentKeywords = ['contentEncoding' => 'base64'];
+    if (NULL !== $field->getMaxFileSize() && $field->getMaxFileSize() > 0) {
+      $contentKeywords['$validations'] = [
+        JsonSchema::fromArray([
+          'keyword' => 'maxLength',
+          // The file might need up to 36 % more space through Base64 encoding.
```

Comment:
The English Wikipedia has a higher bid of up to 37% :wink:

Reply:
Changed :slightly_smiling_face: